src/Microsoft.AspNet.OData.Shared/IEdmChangedObject.cs

```diff
@@ -14,5 +16,15 @@ public interface IEdmChangedObject : IEdmStructuredObject
         /// Used to determine which Delta object to create during serialization.
         /// </summary>
         EdmDeltaEntityKind DeltaKind { get; }
+
+        /// <summary>
+        /// To hold Data Modification Exception
+        /// </summary>
+        DataModificationException DataModificationException { get; set; }
+
+        /// <summary>
+        /// To hold Instance Annotations
+        /// </summary>
+        IODataInstanceAnnotationContainer InstanceAnnotations { get; set; }
```

Comment:
IEdmChangedObject is a public interface that other developers may have implementations of -- we can't add new properties.  We can define an IEdmAnnotatedObject with these properties, and have our classes implement that interface as well.